source/compiler/tests/gh_353.pwn

```diff
@@ -0,0 +1,26 @@
+#include <console>
+#include <string>
+main()
+{
+	new var_01 = 0;		// the compiler shouldn't suggest any name for this error
+	printf("%d", var);	// since "var_01" and "var" differ by more than 2 symbols
+
+	new var = 1;
+	printf("%d", val); // error 017: undefined symbol "val"; did you mean "var"?
+
+	printf("%d", celmax); // error 017: undefined symbol "celmax"; did you mean "cellmax"?
+
+	new str[4] = "a";
+	strcaf(str, "b"); // error 017: undefined symbol "strcaf"; did you mean "strcat"?
+
+lbl:
+	goto lb; // error 019: not a label: "lb"; did you mean "lbl"?
+
+	// error 086: unknown automaton "dialpg"; did you mean "dialog"?
+	state dialpg:DLG_REGISTER;
+
+	// error 087: unknown state "DLGLOGIN" for automaton "dialog"; did you mean "DLG_LOGIN"?
+	state dialog:DLGLOGIN;
+}
+stock DialogHandler() <dialog:DLG_REGISTER> {}
```

Comment:
Would also be nice to have a test case something like:

```pawn
Func1() <state_1:STATE_1>{}
Func1() <state_1:STATE_2>{}

Func2() <state_2:BEING_1>{}
Func2() <state_2:BEING_2>{}

state state_2:STATE_2;
```

Which should NOT suggest an alternative, because `STATE_2` is in a different automata.  The code looks like it does this, but confirmation would be nice.

Or even better, in that case suggest that maybe you meant `state_1:STATE_2`, which the code currently does not do.